src/engine/engineState.js

```diff
@@ -290,6 +290,11 @@ export class EngineState extends EventEmitter {
           if (++bad === uris.length) {
             const msg = e ? `With error ${e.toString()}` : ''
             this.log(`broadcastTx fail: ${rawTx}\n${msg}}`)
+            if (e != null && e.message.indexOf('SpamServerDetected') !== -1) {
```

Comment:
Move this logic to the onmessage handler. The server should be scored down if it ever returns a spam response. Putting the code here would only down score of the server if we are out of other servers.